platform/osgi/platform-osgi-configadmin/src/main/java/org/apache/felix/cm/impl/ConfigurationManager.java

```diff
@@ -173,6 +176,13 @@
   private final HashMap<String, ConfigurationImpl> configurations =
       new HashMap<String, ConfigurationImpl>();
 
+  // (CODICE) Moved the persistence manager to class state to support closeable
+  private WrappedPersistenceManager fpm;
+
+  // (CODICE) Need a lock for the config dead zone where plugins that initialize and query
+  // config admin will miss configs currently being written, but not yet cached
+  private final StampedLock configDeadZoneLock = new StampedLock();
```

Comment:
✏️  Why not use a standard ReadWriteLock? The advantage of that one is that you could give to your class either the write or the read lock based on your design as opposed to have them get the right one. It clarifies the code since they receive a Lock without knowing it is a read or a write one. For them, it is just a lock.

Reply:
I think we get an efficiency boost with this one since it is not reentrant, nor does it need to be. I did consider injecting with the `Lock` interface, but since `StampedLock` does provide convenience methods for direct acquisition I felt it better to show the kind of lock being acquired in the precise place it gets acquired. Thus when someone reads this later, the justification comments will make sense without them having to scroll up to review what's getting injected where.

Reply:
This will be moot now because of your follow-up PR which will concentrate locking in your persistent manager instead of having it done in multiple places.